src/sample/models/Bot.java

```diff
@@ -31,11 +31,101 @@ public void Move(MoveEnum direction)
     public void MoveRandomly()
     {
         // Determine which moves are possible, then add them to the list
+        getPreferredMoves();
         getPossibleMoves();
         Random randomNumberGenerator = new Random(System.currentTimeMillis());
-        int randomMovementNumber = randomNumberGenerator.nextInt(possibleMoves.size());
-        MoveEnum chosenMove = possibleMoves.get(randomMovementNumber);
-        moveTheBot(chosenMove);
+        if (preferredMoves.size() != 0)
+        {
+            int randomMovementNumber = randomNumberGenerator.nextInt(preferredMoves.size());
+            MoveEnum chosenMove = preferredMoves.get(randomMovementNumber);
+            if (possibleMoves.contains(chosenMove)) {
+                moveTheBot(chosenMove);
+            }
+            preferredMoves.clear();
+        }
+        else
+            {
+                int randomMovementNumber = randomNumberGenerator.nextInt(possibleMoves.size());
+                MoveEnum chosenMove = possibleMoves.get(randomMovementNumber);
+                moveTheBot(chosenMove);
+            }
+    }
+
+    private void getPreferredMoves()
+    {
+        if(this.y != 0)
+        {
+            if (movesMade.size() == 0)
+            {
+                preferredMoves.add(MoveEnum.Up);
+            }
+            for (String move : movesMade)
+            {
+                // extracting x and y
+                int xMove = Integer.parseInt(move.split(" ")[0]);
```

Comment:
Are we just checking to see if the square we're trying to move to has been visited?

Reply:
Yeah I can simplify this code a bit I will do that. It was a logic bug so I can remove xMove and yMove now I believe